chia/wallet/puzzles/singleton_top_layer.py

```diff
@@ -185,62 +186,40 @@ def adapt_inner_puzzle_hash_to_singleton(inner_puzzle_hash: bytes32) -> bytes32:
 
 def remove_singleton_truth_wrapper(puzzle: Program) -> Program:
     inner_puzzle = puzzle.rest().first().rest()
-    return inner_puzzle
+    # TODO: Remove cast when we improve typing
+    return cast(Program, inner_puzzle)
```

Comment:
this doesn't have test coverage. it seems a bit scary, is that the right place to find the inner puzzle?

Reply:
Truths are a bit of an outdated concept, this seems to be legacy code for when we were planning on using the concept but the whole function (and its corresponding function in singleton_top_layer_v1_1) can be deleted it seems.

Reply:
Sent PR #17383 for that.